thunder/core/jit_ext.py

```diff
@@ -74,6 +74,10 @@
 from thunder.torch import _torch_to_thunder_function_map
 from thunder.clang import _clang_fn_set
 from thunder.core.pytree import tree_map, tree_iter
+from thunder.torch.experimental.dtensor_torch_and_prims import register_dtensor_torch_and_prims
+
+# TODO: Find a better place to register these ops (mostly in thunder/torch/__init__.py but without cyclical dependency).
+register_dtensor_torch_and_prims()
```

Comment:
nit-picking: Potential necessity of `torch.distributed` check.
Probably I just don't understand the implementation correctly but I'm not convinced with this registration's "safety" when `torch.distributed` is not available because IIUC this registration really uses `DTensor` and run some "meta" computation.

Though in most cases the submodule itself should be available so I'm not that strongly concerned.

Reply:
Yes, even though it's not part of GitHub CI, we need to make sure that Thunder is importable with a PyTorch version that was compiled with the USE_DISTRIBUTED=0 flag. There are NVIDIA-internal CI jobs that use PyTorch without distributed, and they would fail eventually.